src/rpc/rawtransaction.cpp

```diff
@@ -262,13 +299,33 @@ static RPCHelpMan getrawtransaction()
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, errmsg + ". Use gettransaction for wallet transactions.");
     }
 
-    if (!fVerbose) {
+    if (verbosity <= 0) {
         return EncodeHexTx(*tx, RPCSerializationFlags());
     }
 
     UniValue result(UniValue::VOBJ);
-    if (blockindex) result.pushKV("in_active_chain", in_active_chain);
-    TxToJSON(*tx, hash_block, result, chainman.ActiveChainstate());
+    if (blockindex) {
+        result.pushKV("in_active_chain", in_active_chain);
+    }
+
+    CBlockUndo blockUndo;
+    CBlock block;
+    const bool is_block_pruned{WITH_LOCK(cs_main, return chainman.m_blockman.IsBlockPruned(blockindex))};
+
+    if (verbosity == 1 || tx->IsCoinBase() ||
+        !blockindex || is_block_pruned ||
```

Comment:
It seems if `verbosity==2` and `!blockindex` this will trigger and we won't get the `fee` and `prevout` fields even when `g_txindex` is available.. Is this expected behavior?

Reply:
You are correct. I have refactor the code to detect if blockhash was sent as a parameter. 
What should we expect if the blockhash is **not** sent and the tx is in **mempool**? Should the fee field be available? (I would think so)